src/components/SquareBox/SquareBox.tsx

```diff
@@ -1,57 +1,75 @@
+import { Circle } from './Circle';
+import { Square } from './Square';
 
 const styles = StyleSheet.create({
-    main: {
-        display: 'flex',
-        flexDirection: 'column',
-        margin: '25px'
-    },
-    control: {
-        width: '100%',
-        display: 'flex',
+  main: {
+    display: 'flex',
+    flexDirection: 'column',
+    margin: '25px',
+  },
+  control: {
+    width: '100%',
+    display: 'flex',
+  },
+  squares: {
+    marginTop: '25px',
+    display: 'flex',
+    child: {
+      marginRight: '10px',
     },
-    squares: {
-        marginTop: '25px',
-        display: 'flex',
-        'child': {
-            marginRight: '10px'
-        }
-    },
-    square: {
-        marginRight: '15px'
-    }
+  },
+  square: {
+    marginRight: '15px',
+  },
 });
 
 export function SquareBox() {
-    const id = React.useId();
-    const [squares, setSquares] = React.useState<Array<string>>([]);
-
-    const deleteSquare = function () {
-        squares.pop();
-        setSquares(squares => [...squares]);
-    };
+  interface Shape {
+    color: string;
+    shape: string;
+  }
+  const id = React.useId();
+  const [squares, setSquares] = React.useState<Array<Shape>>([]);
+  const addCircle = () => {
+    setSquares((squares) => ([
+      ...squares, { color: randomElement(Object.values(COLORS)), shape: 'circle' },
+    ]
+    ));
+  };
 
-    const addSquare = React.useCallback(
-        () => {
-            setSquares(squares => ([
-                ...squares,
-                randomElement(Object.values(COLORS))]
-            ))
-        }, []);
+  const deleteSquare = () => {
+    squares.pop();
+    setSquares((squares) => [...squares]);
+  };
+  const addSquare = React.useCallback(
+    () => {
+      setSquares((squares) => ([
+        ...squares,
+        { color: randomElement(Object.values(COLORS)), shape: 'square' }]
+      ));
+    },
+    [],
+  );
 
-    return <div className={css(styles.main)}>
-        <div className={css(styles.control)}>
-            <Button onClick={deleteSquare} variant='contained'>REMOVE SQUARE</Button>
-            <Button onClick={addSquare} variant='contained'>Add square</Button>
-        </div>
+  return (
+    <div className={css(styles.main)}>
+      <div className={css(styles.control)}>
+        <Button onClick={deleteSquare} variant="contained">REMOVE SQUARE</Button>
+        <Button onClick={addSquare} variant="contained">Add square</Button>
+        <Button onClick={addCircle} variant="contained">Add Circle</Button>
+      </div>
 
-        <div className={css(styles.squares)}>
-            {squares.map((square) => <Square key={id} color={square} />)}
-        </div>
-    </div>;
-}
\ No newline at end of file
+      <div className={css(styles.squares)}>
+        {squares.map((square: Shape, id) => (square.shape === 'square'
+          ? <Square key={id} color={square.color} /> : square.shape === 'circle'
+            ? <Circle key={id} color={square.color} /> : 0))}
```

Comment:
```suggestion
        {squares.map((square: Shape) => square.shape === 'square'
          ? <Square key={id} color={square.color} /> : <Circle key={id} color={square.color} />)}
```

We have to options so we can check only one shape and use the other as default.

Use should use `id` from line 37, not the index from map function.